kedro-datasets/kedro_datasets_experimental/xarray/geotiff_dataset.py

```diff
@@ -0,0 +1,133 @@
+        ...     dims=("band", "y", "x"),
+        ...     coords={"band": [1, 2], "y": [0.5, 1.5, 2.5], "x": [0.5, 1.5]}
+        ... )
+        >>> data_crs = data.rio.write_crs("epsg:4326")
+        >>> data_spatial_dims = data_crs.rio.set_spatial_dims("x", "y")
+        >>> dataset = GeoTiffDataset(filepath="test.tif")
+        >>> dataset.save(data_spatial_dims)
+        >>> reloaded = dataset.load()
+        >>> xr.testing.assert_allclose(data_spatial_dims, reloaded, rtol=1e-5)
+
+    """
+
+    DEFAULT_LOAD_ARGS: dict[str, Any] = {}
+    DEFAULT_SAVE_ARGS: dict[str, Any] = {}
+
+    def __init__(  # noqa: PLR0913
+        self,
+        filepath: str,
+        load_args: dict[str, Any] = None,
+        save_args: dict[str, Any] = None,
+        version: Version = None,
+        metadata: dict[str, Any] = None,
+    ):
+        """Creates a new instance of ``GeoTiffDataset`` pointing to a concrete
+        tiff or tif file with geospatial data.
+
+        Args:
+            filepath: Filepath in POSIX format to a geotiff file.
+                The prefix should be any protocol supported by ``fsspec``.
+            load_args: rioxarray options for loading geotiff files.
+                Here you can find all available arguments:
+                https://corteva.github.io/rioxarray/html/rioxarray.html#rioxarray-open-rasterio
+                All defaults are preserved.
+            save_args: rioxarray options for saving to raster file in geotiff format.
+                Here you can find all available arguments:
+                https://corteva.github.io/rioxarray/html/rioxarray.html#rioxarray.raster_dataset.RasterDataset.to_raster
+                All defaults are preserved, but "index", which is set to False.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            metadata: Any arbitrary metadata.
+                This is ignored by Kedro, but may be consumed by users or external plugins.
+        """
+        protocol, path = get_protocol_and_path(filepath, version)
+        self._protocol = protocol
+        self._fs = fsspec.filesystem(self._protocol)
+        self._format = "GEOTIFF"
+        self.metadata = metadata
+
+        super().__init__(
+            filepath=PurePosixPath(path),
+            version=version,
+            exists_function=self._fs.exists,
+            glob_function=self._fs.glob,
+        )
+
+        # Handle default load and save arguments
+        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+    def _load(self) -> xarray.DataArray:
```

Comment:
It's also nice to support multiple files if you want to concatenate many individual layers into a multi-dimensional xarray object. See how we do it for NetCDF here: https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/netcdf/netcdf_dataset.py.

```python
# Note that we require an explicit "concat_dim" entry in the dataset
# catalog. There might be a better way to do this though.
concat_dim = self._load_args.get("concat_dim", None)

remote_files = fs.glob(load_path)
if concat_dim:
    preprocess = partial(
        self._preproc_multiple_geotiff, concat_dim=concat_dim
    )
data = xr.open_mfdataset(
    remote_files,
    combine_attrs="drop_conflicts",
    engine="rasterio",
    **self._load_args,
).sortby(coords)

def _preproc_multiple_geotiff(self, x: xr.DataArray, concat_dim: str):
    """Helper function for open_mfdataset to set up coordinates for concatenated
       geotiffs.

   `concat_dim` is a required argument in the DataCatalog and should also be an
    attribute/tag on the geotiff for labeling the coordinate.

    Args:
        x: Individual tif DataArray being concatenated (handled automatically by
            `open_mfdataset`).
        concat_dim: Name of the dimension to concatenate over.

    Returns:
        Individual DataArray slice to be concatenated with proper formatting,
        coordinates, attributes, etc.
    """
    # Squeeze out `band_data`
    x = x.squeeze()

    # Pull individual filepath so we can use `rasterio` to get the tag
    # attributes.
    path = x.encoding["source"]
    with rasterio.open(path) as data:
        # Pull in relevant tags like units and return period.
        tags = data.tags()
    
    # Note here that this makes this tedious as well. Need to have `concat_dim`
    # in the catalog as well as a tag in the tif to label the axis. I guess you
    # could not require this and would just leave the xarray coordinate unlabeled
    if concat_dim not in tags:
        raise ValueError(
            f"{concat_dim} needs to be a tag in geotiff files to help with "
            + "coordinate axis labeling."
        )
    x.attrs.update(tags)

    x.expand_dims(concat_dim)
    x = x.assign_coords({concat_dim: tags[concat_dim]})
    return x
```